### PR TITLE
seo: Add SEO meta tags for CMS and Pi.

### LINKED ISSUE #2369: OpenGraph tags
There are no OpenGraph tags in the politeiagui HTML, which means that links shared on social media do not have any image, logo or description.

The following tools can be used to preview how shared links will appear on Twitter and Facebook.

https://cards-dev.twitter.com/validator
https://developers.facebook.com/tools/debug/?q=proposals.decred.org

Compare with the preview of dex.decred.org to see the potential improvement.

### LINKED ISSUE #2600: Improve search engine result snippet
When you search `decred politeia` in a search engine, the following result is displayed.

image

We need to fix these results.

https://developers.google.com/search/docs/advanced/appearance/good-titles-snippets

### PR DESCRIPTION
**Pull request by victorgcramos**
Closes #2600
Closes #2369

This diff adds SEO meta tags for search engines, Facebook and
Twitter. With these tags, we should be able to provide better results
when searching for "decred politeia", or other keywords, and display a
better preview on Twitter and Facebook posts, including image, app
title and description.

### CODE REVIEW

File: config/politeia.env [hunk 1]
```diff
@@ -1,3 +1,7 @@
 HTTPS=true
 REACT_APP_PRESET=politeia
 REACT_APP_WEBSITE_NAME=Politeia
+REACT_APP_WEBSITE_SUBTITLE="Decred's Governance System"
```
Comment (lukebp) [inline config/politeia.env:4]:
```suggestion
REACT_APP_WEBSITE_SUBTITLE="Decred's Proposal System"
```

File: public/index.html [hunk 2]
```diff
@@ -1,24 +1,70 @@
...
+      name="twitter:title"
+      content="%REACT_APP_WEBSITE_NAME% - %REACT_APP_WEBSITE_SUBTITLE%"
+    />
+    <meta
+      name="twitter:description"
+      content="%REACT_APP_WEBSITE_DESCRIPTION%"
+    />
+    <meta
+      name="twitter:image"
+      content="%PUBLIC_URL%/assets/images/%REACT_APP_WEBSITE_BANNER_FILENAME%"
```
Comment (lukebp) [inline public/index.html:36]: I deployed this on test-proposals to test it. The twitter card image is not coming through.

image

File: config/politeia.env [hunk 3]
```diff
@@ -1,3 +1,7 @@
...
+REACT_APP_WEBSITE_DESCRIPTION="A platform that supports Decred’s governance. It facilitates the submission, tracking, and discussion of Decred governance proposals by the stakeholder community."
```
[the remark this answers is not in the document]
Comment (victorgcramos) [reply config/politeia.env:5]: oh, okay! Will change that!

[the remark this answers is not in the document]
Comment (victorgcramos) [reply config/politeia.env:5]: Could you give a description with less than 170 chars? The SEO recommendation is to have at most 170.

[the remark this answers is not in the document]
Comment (lukebp) [reply config/politeia.env]: 160 chars

Politeia is Decred's proposal system. It's used to request funding from the Decred treasury. The Decred stakeholders decide how treasury funds are allocated.

[the remark this answers is not in the document]
Comment (lukebp) [reply config/politeia.env]: Updated it a few times, so make sure you get the most recent edit.

**Review by lukebp — CHANGES_REQUESTED** [review]
The Politeia image is still not being shown.  Everything else looks good.

image

**Comment by victorgcramos** [comment]
Here's the preview for Twitter and Facebook:

Screen Shot 2021-09-29 at 7 12 58 PM 
Screen Shot 2021-09-29 at 7 19 41 PM

In order to have better images for both, we need to deploy new files to production, since the SEO bots does not render javascript, hence metatag urls must be static.

**Review by lukebp — APPROVED** [review]
tACK

image

**Comment by lukebp** [comment]
Before
image

After
image

**Pull request merged**